### PR TITLE
ref(auth): Clean up auth provider feature flags

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- Allow providers to specify what feature they require
 - Verify a provider is enabled during setup
 - Adds sso-rippling as it's own feature

### CODE REVIEW

File: docs/sso.rst [hunk 1]
```diff
@@ -20,6 +20,13 @@ with a feature switch in your ``sentry.conf.py``::
     # turn SSO on our off
     SENTRY_FEATURES['organizations:sso'] = False
 
+Additionally you may enabled advanced SSO features::
```
Comment (dcramer) [inline docs/sso.rst]: enabled -> enable

File: src/sentry/auth/providers/saml2.py [hunk 2]
```diff
@@ -226,6 +226,8 @@ class SAML2Provider(Provider):
       state during setup. The attribute mapping should map the `Attributes`
       constants to the Identity Provider attribute keys.
     """
+    # Most SAML providers require the advanced SSO feature
+    required_feature = 'organizations:sso-adv'
```
Comment (dcramer) [inline src/sentry/auth/providers/saml2.py]: on second thought, what if we kept this named 'sso-saml2'

Comment (EvanPurkhiser) [reply src/sentry/auth/providers/saml2.py]: Hmm, yeah, ok I think I agree with that.

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Changes require @getsentry/security sign-off</td>
    </tr>
  </tbody>
</table>

### Security concerns found

- src/sentry/auth/provider.py
- src/sentry/auth/providers/saml2.py
- src/sentry/web/frontend/organization_auth_settings.py


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by EvanPurkhiser** [comment]
Also going to throw matt on this for review.

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**